src/main/java/org/testng/DataProviderHolder.java

```diff
@@ -0,0 +1,44 @@
+package org.testng;
+
+import java.util.Collection;
+import java.util.Collections;
+import org.testng.collections.Sets;
+
+/**
+ * A holder class that is aimed at acting as a container for holding various different aspects of a
+ * data provider such as listeners/interceptors etc.,
+ */
+public class DataProviderHolder {
+
+  private Collection<IDataProviderListener> listeners = Sets.newHashSet();
```

Comment:
could be final

Reply:
Sure. I will fix this. @juherr - Any other observations or can I consider this approved and can proceed with getting this merged ?

Reply:
no, everything else looks good 👍